cmd/tellor/main.go

```diff
@@ -233,135 +250,145 @@ func newDisputeCmd(cmd *cli.Cmd) {
 	}
 }
 
-func mineCmd(cmd *cli.Cmd) {
-	remoteDS := cmd.BoolOpt("remote r", false, "connect to remote dataserver")
-	cmd.Action = func() {
-		// Create os kill sig listener.
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, os.Interrupt)
-		exitChannels := make([]*chan os.Signal, 0)
-
-		cfg := config.GetConfig()
-		var ds *ops.DataServerOps
-		if !cfg.EnablePoolWorker {
-			ErrorHandler(AddDBToCtx(*remoteDS), "initializing database")
-			if !*remoteDS {
-				ch := make(chan os.Signal)
-				exitChannels = append(exitChannels, &ch)
-
-				var err error
-				ds, err = ops.CreateDataServerOps(ctx, ch)
-				if err != nil {
-					log.Fatal(err)
-				}
-				// Start and wait for it to be ready.
-				if err := ds.Start(ctx); err != nil {
-					log.Fatal(err)
+func mineCmd(logger log.Logger) func(*cli.Cmd) {
+	return func(cmd *cli.Cmd) {
+		remoteDS := cmd.BoolOpt("remote r", false, "connect to remote dataserver")
+		cmd.Action = func() {
+			// Create os kill sig listener.
+			c := make(chan os.Signal, 1)
+			signal.Notify(c, os.Interrupt)
+			exitChannels := make([]*chan os.Signal, 0)
+
+			cfg := config.GetConfig()
+			var ds *ops.DataServerOps
+			if !cfg.EnablePoolWorker {
+				ErrorHandler(AddDBToCtx(*remoteDS), "initializing database")
+				if !*remoteDS {
+					ch := make(chan os.Signal)
+					exitChannels = append(exitChannels, &ch)
+
+					var err error
+					ds, err = ops.CreateDataServerOps(ctx, logger, ch)
+					if err != nil {
+						level.Error(logger).Log("msg", "error creating data server", "err", err)
+						os.Exit(1)
+					}
+					// Start and wait for it to be ready.
+					if err := ds.Start(ctx, logger); err != nil {
+						level.Error(logger).Log("msg", "error starting data server", "err", err)
+					}
+					<-ds.Ready()
 				}
-				<-ds.Ready()
 			}
-		}
-		// Start miner
-		DB := ctx.Value(tellorCommon.DataProxyKey).(db.DataServerProxy)
-		v, err := DB.Get(db.DisputeStatusKey)
-		if err != nil {
-			fmt.Println("ignoring --- could not get dispute status.  Check if staked")
-		}
-		status, _ := hexutil.DecodeBig(string(v))
-		if status.Cmp(big.NewInt(1)) != 0 {
-			log.Fatalf("Miner is not able to mine with status %v. Stopping all mining immediately", status)
-		}
-		ch := make(chan os.Signal)
-		exitChannels = append(exitChannels, &ch)
-		miner, err := ops.CreateMiningManager(ctx, ch, ops.NewSubmitter())
-		if err != nil {
-			log.Fatal(err)
-		}
-		miner.Start(ctx)
-
-		// Wait for kill sig.
-		<-c
-		// Then notify exit channels.
-		for _, ch := range exitChannels {
-			*ch <- os.Interrupt
-		}
-		cnt := 0
-		start := time.Now()
-		for {
-			cnt++
-			dsStopped := false
-			minerStopped := false
-
-			if ds != nil {
-				dsStopped = !ds.Running
-			} else {
-				dsStopped = true
+			// Start miner
+			DB := ctx.Value(tellorCommon.DataProxyKey).(db.DataServerProxy)
+			v, err := DB.Get(db.DisputeStatusKey)
+			if err != nil {
+				level.Warn(logger).Log("info", "could not get dispute status. Check if staked")
```

Comment:
```suggestion
				level.Warn(logger).Log("msg", "could not get dispute status. Check if staked")
```

lets keep the format the same everywhere. See prometheus and thanos for examples.
when you have `level.Info` it automatically adds INFO to the logs. the same for warn and error logs